helpers/githubHelper.js

```diff
@@ -1,52 +1,100 @@
-			.send(this.createPullRequestRequestBody(head, title, body))
-			.set('Authorization', `token ${accessToken}`)
-			.end((err, dataToReturn) => {
-				callback(null, dataToReturn);
-			});
+
+	createPullRequest(head, title, body, accessToken) {
+		return new Promise( (resolve) => {
+			this.agent()
+				.post(this.config().createPullRequest)
+				.send(this.createPullRequestRequestBody(head, title, body))
+				.set('Authorization', `token ${accessToken}`)
+				.end((err, dataToReturn) => {
+					resolve(dataToReturn);
+				});
+		});
 	}
+
 	createContentRequestBody(testFileName, branchName, commitMessage, base64FileContents) {
 		return `{"path":"${testFileName}", "branch":"${branchName}", "message":"${commitMessage}", "content":"${base64FileContents}"}`;
 	}
-	createContent(accessToken, testFileName, branchName, commitMessage, base64FileContents, callback) {
-		this.agent()
-			.put(`${this.config().github.apiEndpoints.createContent}${testFileName}`)
-			.send(this.createContentRequestBody(testFileName, branchName, commitMessage, base64FileContents))
-			.set('Authorization', `token ${accessToken}`)
-			.end((err, createCommitResult) => {
-				callback(null, createCommitResult);
-			});
+
+	createContent(accessToken, testFileName, branchName, commitMessage, base64FileContents) {
+		return new Promise( (resolve) => {
+			this.agent()
+				.put(`${this.config().createContent}${testFileName}`)
+				.send(this.createContentRequestBody(testFileName, branchName, commitMessage, base64FileContents))
+				.set('Authorization', `token ${accessToken}`)
+				.end((err, createCommitResult) => {
+					resolve(createCommitResult);
+				});
+		});
 	}
+
 	createReferenceRequestBody(newBranchName, commitReferenceToBranchFrom) {
 		return `{"ref":"refs/heads/${newBranchName}", "sha":"${commitReferenceToBranchFrom}"}`;
 	}
-	createReference(accessToken, newBranchName, commitReferenceToBranchFrom, callback) {
-		this.agent()
-			.post(this.config().github.apiEndpoints.createRef)
-			.send(this.createReferenceRequestBody(newBranchName, commitReferenceToBranchFrom))
-			.set('Authorization', `token ${accessToken}`)
-			.end((err, createReferenceResult) => {
-				callback(null, createReferenceResult);
-			});
+
+	createReference(accessToken, newBranchName, commitReferenceToBranchFrom) {
+		return new Promise( (resolve) => {
+			this.agent()
+				.post(this.config().referencesEndpoint)
+				.send(this.createReferenceRequestBody(newBranchName, commitReferenceToBranchFrom))
+				.set('Authorization', `token ${accessToken}`)
+				.end((err, createReferenceResult) => {
+					resolve(createReferenceResult);
+				});
+		});
+	}
+
+	getHeadReferenceForBranch(requestedBranch) {
+		return new Promise( (resolve, reject) => {
+			this.agent()
+				.get(this.config().referencesEndpoint)
+				.end((err, response) => {
+
+					const responseBody = response.body;
+					if (responseBody && Array.isArray(responseBody)) {
+						let branchRef = null;
```

Comment:
No particular reason appart from old habbits that tend to die hard. Sould be fixed in 621876ee8b03b6bd830b5ec39c5ce1293f760a23